### ISSUE TITLE
Replace PHP Templates

### ISSUE THREAD

**Issue by devinrayolsen** [connectivedx/Phoenix#188]
@elseloop Lets use this ticket to keep an on going conversation about PHP template shift. 

I know we are working on a Handlebars version to replace our PHP templates, but I thought it might be worth while to show how Jade template engine would work (https://github.com/connectivedx/Phoenix/tree/feature/jade-templates). Jade is still widely used and actually comes with the Node Express for application framework.

https://www.npmjs.com/package/gulp-jade

- New html build driver setup to accept jade API configuration directly from gulp file.
- Global and local vars with an options.jade to condition them if needed.
- Layouts are now properly divided out from templates and extend from.
- Produces html files. did I mention that yet?
- Includes are welcome
- MIXINS!

Code references = http://jade-lang.com/reference/
Jade gulp file configuration = http://jade-lang.com/api/

And discuses...

**Comment by ajmueller** [comment]
I've been using Jade on a lot of projects lately and LOVE it.  Plus it's [being renamed to Pug](https://github.com/pugjs) and who doesn't love pugs?!

I think the only concern might be how back-end devs would be able to digest the compiled HTML templates.  Having PHP templates broken out into easily digestible pieces that line up well with Sitecore or other CMS components makes it easy for them to copy/paste into their environment.  Using Jade, they'd have to have a way to know how they need to be broken up (which may very well be handled perfectly by using HTML comments similar to how @kamsar uses HTML comments to separate Sitecore components).

**Comment by devinrayolsen** [comment]
@ajmuller Good point. I did add a /inc/components/ folder for us to house our different include components to our templates (I think something we are trying to do more often now), however I too wonder if these would be easily digestible for backed implementations.

**Comment by ajmueller** [comment]
@devinrayolsen ah, very nice.  Since the compiled HTML results in those components being incorporated into other HTML, though, the best way (I would imagine) to help back-end devs would be to add `'./templates/inc/**/*.jade'` to the paths property of the HTML rendering config option.  This, along with comments in the Jade templates to help with context, would allow back-end devs to consume the markup without having to learn Jade's syntax (plus they can copy/paste as before).

**Comment by devinrayolsen** [comment]
@ajmuller Great point, thanks! Updated. 
Any suggestions on getting production and dev builds to run this new html driver? 
Looks like only watch is available.

**Comment by ajmueller** [comment]
@devinrayolsen what do you mean by "top level files?"  Are you referring to the templates at the top level of the templates directory vs. those in the `inc` directory?  I deleted the `dist` directory and ran the production and dev builds and everything looked good.

- all .jade files immediately located in `src/templates` (and not in sub-directories) were compiled and moved to `dist/`
- all .jade files in `src/templates/inc` were compiled and moved to `dist/inc/`

**Comment by devinrayolsen** [comment]
Sorry by top level I mean root of the checkout along side /Assets and /Documentation.
I could be wrong but I thought dev build at the very least (like build watch is doing now) would create html files and /inc folder at root of checkout. No?

**Comment by ajmueller** [comment]
Oh right, so by top level you mean how any other human would interpret top level.  My bad!

That's a good question and it appears that it's supposed to be doing that according to the output property.  It does do that with the current PHP output.  Perhaps @kamsar could enlighten us.

**Comment by kamsar** [comment]
You got me, it's been a while since I wrote that stuff.
